compiler/rustc_codegen_llvm/src/builder.rs

```diff
@@ -462,15 +462,14 @@ impl BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
             load: &'ll Value,
             scalar: &abi::Scalar,
         ) {
-            let vr = scalar.valid_range;
             match scalar.value {
                 abi::Int(..) => {
                     let range = scalar.valid_range_exclusive(bx);
                     if range.start != range.end {
                         bx.range_metadata(load, range);
                     }
                 }
-                abi::Pointer if vr.start < vr.end && !vr.contains(0) => {
```

Comment:
I have implemented this as a generic `contains_zero` method.
Where I check for `vr.start <= vr.end`, is there any reason that `start` and `end` couldn't be equal when checking for what I think is NonNull?
`WrappingRange{start:1, end: 1}.contains_zero()` should be true is my assumption.

Reply:
> I have implemented this as a generic contains_zero method.

What is the difference to `.contains(0)`?